app/team.go

```diff
@@ -183,6 +183,24 @@ func (a *App) UpdateTeamScheme(team *model.Team) (*model.Team, *model.AppError)
 	return oldTeam, nil
 }
 
+func (a *App) UpdateTeamPrivacy(team *model.Team) (*model.Team, *model.AppError) {
```

Comment:
This essentially is a limited copy of the `UpdateTeam` method currently, I think if the API would accept only the parameters explicitly (string, boolean) it would make more sense than a struct. Otherwise you need to create a struct exclusively to call this method - Then you could also call the update or patch methods. - Thoughts @jespino ?

Reply:
@DSchalla What are the (string,boolean) stands for? Doesn't it just need the team name if we want to specify which team it is?

Any thought on this, @jespino ?

Reply:
Yes, I think I agree with @DSchalla here. probably changing the function to something like:

```go
func (a *App) UpdateTeamPrivacy(teamId string, teamType string, allowOpenInvite boolean) (*model.Team, *model.AppError) {
```

should work properly. and because we are not using the returned team, maybe we can just return `*model.AppError`.